pkg/maps/ctmap/cell.go

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package ctmap
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/cilium/hive/cell"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/maps/nat"
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/option"
+)
+
+// Cell provides the ctmap.Map which contains the connection tracking state.
+var Cell = cell.Module(
+	"ct-map",
+	"eBPF map which manages connection tracking",
+
+	cell.Provide(newCTMaps),
+)
+
+func newCTMaps(lifecycle cell.Lifecycle, daemonConfig *option.DaemonConfig, registry *metrics.Registry, natMap4 nat.NatMap4, natMap6 nat.NatMap6) bpf.MapOut[CTMaps] {
+	InitMapInfo(natMap4, natMap6)
+
+	ctMaps := &ctMaps{}
+
+	if daemonConfig.IPv4Enabled() {
+		ctMaps.v4AnyMap = newMap(MapNameAny4Global, mapTypeIPv4AnyGlobal, registry)
+		ctMaps.v4TCPMap = newMap(MapNameTCP4Global, mapTypeIPv4TCPGlobal, registry)
+	}
+
+	if daemonConfig.IPv6Enabled() {
+		ctMaps.v6AnyMap = newMap(MapNameAny6Global, mapTypeIPv6AnyGlobal, registry)
+		ctMaps.v6TCPMap = newMap(MapNameTCP6Global, mapTypeIPv6TCPGlobal, registry)
+	}
+
+	lifecycle.Append(cell.Hook{
+		OnStart: func(context cell.HookContext) error {
+			return ctMaps.init()
+		},
+		OnStop: func(context cell.HookContext) error {
+			return ctMaps.close()
+		},
+	})
+
+	return bpf.NewMapOut(CTMaps(ctMaps))
+}
+
+// CTMaps provides access to the active connection tracking BPF maps.
+type CTMaps interface {
```

Comment:
nit: best I can tell this is only used in the ct/gc package. Better to define the interface where it's actually used.